### PR TITLE
chore: add kfp-kubernetes docs and update process infrastructure

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
* Adds documentation infra/layout
* Updated docstrings for clarity/readability
* Orders package objects logically in docs
* Add two scripts to automate/document release process: `create_release_branch.sh` and `release.sh`
    * This follows the same docs release process as GCPC: create a release branch, overwrite the `.readthedocs.yml` file in the repo root, release from the branch, push the branch, and build the docs from the release branch
    * `.readthedocs.yml` needs to be copied to root in the release branch because readthedocs only supports one docs build per repo, which is currently occupied by the KFP SDK in master
* Updates test script so that tests pass CI

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title 
convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: kubernetes_platform/python/docs/Makefile [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Copyright 2019 The Kubeflow Authors
```
Comment (chensun) [inline kubernetes_platform/python/docs/Makefile:1]: nit: 2023

File: kubernetes_platform/python/docs/conf.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+# html_sidebars = {}
+
+# -- Options for HTMLHelp output ---------------------------------------------
+
+# Output file base name for HTML help builder.
+htmlhelp_basename = 'KfpKubernetesdoc'
+
+# -- Options for LaTeX output ------------------------------------------------
+
+latex_elements = {
```
Comment (chensun) [inline kubernetes_platform/python/docs/conf.py:188]: Can this be removed if we don't ever need latex?

File: kubernetes_platform/python/release.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# run from within ./kubernetes_platform/python
```
Comment (chensun) [inline kubernetes_platform/python/release.sh:1]: Adding `#!/bin/bash` and copyright info to the file?

**Review by chensun — COMMENTED** [review]
/lgtm

File: kubernetes_platform/python/release.sh (hunk 3 above)
Comment (connor-mccarthy) [reply kubernetes_platform/python/release.sh:17]: Done.

File: kubernetes_platform/python/docs/Makefile (hunk 1 above)
Comment (connor-mccarthy) [reply kubernetes_platform/python/docs/Makefile]: Done.

File: kubernetes_platform/python/docs/conf.py (hunk 2 above)
Comment (connor-mccarthy) [reply kubernetes_platform/python/docs/conf.py]: Yes, good thinking. Done.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**